src/h5web/providers/hsds/utils.test.ts

```diff
@@ -0,0 +1,172 @@
+      });
+    });
+
+    it('should convert float types', () => {
+      const leFloatType: HsdsBaseType = {
+        class: HDF5TypeClass.Float,
+        base: 'H5T_IEEE_F32LE',
+      };
+      expect(convertHsdsType(leFloatType)).toEqual({
+        class: HDF5TypeClass.Float,
+        size: 32,
+        endianness: 'LE',
+      });
+
+      const beFloatType: HsdsBaseType = {
+        class: HDF5TypeClass.Float,
+        base: 'H5T_IEEE_F64BE',
+      };
+      expect(convertHsdsType(beFloatType)).toEqual({
+        class: HDF5TypeClass.Float,
+        size: 64,
+        endianness: 'BE',
+      });
+    });
+
+    it('should convert the base of Array type', () => {
+      const integerType: HsdsBaseType = {
+        class: HDF5TypeClass.Integer,
+        base: 'H5T_STD_I8LE',
+      };
+      const arrayType: HsdsArrayType = {
+        class: HDF5TypeClass.Array,
+        base: integerType,
+        dims: [4, 5],
+      };
+      expect(convertHsdsType(arrayType)).toEqual({
+        class: HDF5TypeClass.Array,
+        base: {
+          class: HDF5TypeClass.Integer,
+          size: 8,
+          endianness: 'LE',
+        },
+        dims: [4, 5],
+      });
+    });
+
+    it('should convert the base of VLen type', () => {
+      const integerType: HsdsBaseType = {
+        class: HDF5TypeClass.Integer,
+        base: 'H5T_STD_I8LE',
+      };
+      const vlenType: HsdsVLenType = {
+        class: HDF5TypeClass.VLen,
+        base: integerType,
+      };
+      expect(convertHsdsType(vlenType)).toEqual({
+        class: HDF5TypeClass.VLen,
+        base: {
+          class: HDF5TypeClass.Integer,
+          size: 8,
+          endianness: 'LE',
+        },
+      });
+    });
+
+    it('should convert the field types of Compound type', () => {
+      const floatType: HsdsBaseType = {
+        class: HDF5TypeClass.Float,
+        base: 'H5T_IEEE_F64LE',
+      };
+      const integerType: HsdsBaseType = {
+        class: HDF5TypeClass.Integer,
+        base: 'H5T_STD_I8LE',
+      };
+      const vlenType: HsdsVLenType = {
+        class: HDF5TypeClass.VLen,
+        base: integerType,
+      };
+      const compoundType: HsdsCompoundType = {
+        class: HDF5TypeClass.Compound,
+        fields: [
+          {
+            name: 'f1',
+            type: floatType,
+          },
+          {
+            name: 'f2',
+            type: vlenType,
+          },
+        ],
+      };
+      expect(convertHsdsType(compoundType)).toEqual({
+        class: HDF5TypeClass.Compound,
+        fields: [
+          {
+            name: 'f1',
+            type: {
+              class: HDF5TypeClass.Float,
+              size: 64,
+              endianness: 'LE',
+            },
+          },
+          {
+            name: 'f2',
+            type: {
+              class: HDF5TypeClass.VLen,
+              base: {
+                class: HDF5TypeClass.Integer,
+                size: 8,
+                endianness: 'LE',
+              },
```

Comment:
You could probably reuse some of the expected `base` types from previous tests by storing them in constants at the top, like we do in other test files.